src/partition/repart.c

```diff
@@ -4753,6 +4938,24 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_CERTIFICATE_FILE: {
+                        _cleanup_(erase_and_freep) char *cert = NULL;
+                        size_t n = 0;
+
+                        r = read_full_file_full(
+                                        AT_FDCWD, optarg, UINT64_MAX, SIZE_MAX,
+                                        READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
+                                        NULL,
+                                        &cert, &n);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to read certificate file '%s': %m", optarg);
+
+                        erase_and_free(arg_certificate);
+                        arg_certificate = TAKE_PTR(cert);
+                        arg_certificate_size = n;
```

Comment:
maybe parse this right away, following the aforementioned logic of storing stuff in binary form internally. i.e. d2i_X509() the data you just read.

You need to convert from DER anyway for the fingerprint, hence you need to parse the thing anyway. 

We already have pubkey_fingerprint() btw, which might be useful as blueprint for calculating the fingerprint. It operates on just public keys instead of full certificates though. But I guess an x509_fingerprint() could be added that mirrors what it does but takes an X509 object instead of an EVP_PKEY object.